serverside-javascript-best-practices.md

```diff
@@ -680,6 +680,65 @@ async.map(['http://foo', 'http://bar'], dbInfo, ...);
 
 
 
+## Functional JavaScript
+
+JavaScript's
+[first-class functions](https://en.wikipedia.org/wiki/First-class_function)
+enable it to support the
+[functional programming paradigm](https://en.wikipedia.org/wiki/Functional_programming).
+This is _a good thing_: making your code **more functional** can improve its
+readability, testability and maintainability.
+
+### Partial (function) application
+
+Consider the following function:
+
+```javascript
+function subtract(x, y) {
+  return x - y;
+}
+```
+
+The following are all equivalent ways to implement the `subtractFrom5` and
+`subtract5` functions using `subtract`:
+
+* Wrapper functions:
+
+  ```javascript
+var subtractFrom5 = function (x) {
+  return subtract(5, x);
```

Comment:
There is a problem with the indentation here, when he markdown is viewed on github.